osbenchmark/builder/launchers/local_process_launcher.py

```diff
@@ -0,0 +1,187 @@
+import logging
+import os
+import subprocess
+
+import psutil
+
+from osbenchmark import time, telemetry
+from osbenchmark.builder import java_resolver, cluster
+from osbenchmark.builder.launchers.launcher import Launcher
+from osbenchmark.exceptions import LaunchError
+from osbenchmark.utils import io, opts
+from osbenchmark.utils.periodic_waiter import PeriodicWaiter
+
+
+class LocalProcessLauncher(Launcher):
+    PROCESS_WAIT_TIMEOUT_SECONDS = 90
+    PROCESS_WAIT_INTERVAL_SECONDS = 0.5
+
+    def __init__(self, pci, shell_executor, metrics_store, clock=time.Clock):
+        super().__init__(shell_executor)
+        self.logger = logging.getLogger(__name__)
+        self.pci = pci
```

Comment:
I think `provision_config_instance` is better than using `pci` since it's more descriptive.

Reply:
I've gone back and forth on this in my mind a couple times. I settled on `pci` for brevity since many of the accesses to this object will be something like `self.pci.variables["abcd"]["efgh"]`.

But ultimately clarity > brevity so I will refactor. I have used `pci` in a few of the other PRs as well so I will create a followup PR after this one to replace all of the references in 1 go